app/controllers/v0/health_records_controller.rb

```diff
@@ -27,11 +31,19 @@ def create
     def show
       # doc_type will default to 'pdf' if any value, including nil is provided.
       doc_type = params[:doc_type] == 'txt' ? 'txt' : 'pdf'
-      resource = client.get_download_report(doc_type)
-
-      send_data resource.body,
-                type: resource.response_headers['content-type'],
-                disposition: resource.response_headers['content-disposition']
+      response_headers = Hash[REPORT_HEADERS.map { |h| [h, ''] }]
+      chunk_stream = Enumerator.new do |stream|
+        streaming_client.get_download_report(doc_type, response_headers, stream)
+      end
+      first = true
+      chunk_stream.each do |c|
+        if first
+          response_headers.each { |k, v| response[k] = v }
+          first = false
```

Comment:
@patrickvinograd is there a way to determine first by using chunk_stream.each_with_index? where index == 0 or 1